### PR TITLE
[ISSUE #4658] remove redundant logic

### LINKED ISSUE #4658: remove redundant logic in checkTransactionState
- in `DefaultMQProducerImpl#checkTransactionState`, there is redundant if logic, need to be removed

image

- fix getacl config desc

### PR DESCRIPTION
**Pull request by zhiliatom**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

fix  #4658

### CODE REVIEW

**Review by Oliverwqcwrw — DISMISSED** [review]
LGTM

File: tools/src/main/java/org/apache/rocketmq/tools/command/acl/GetAccessConfigSubCommand.java [hunk 1]
```diff
@@ -52,10 +52,10 @@ public class GetAccessConfigSubCommand implements SubCommand {
     @Override public Options buildCommandlineOptions(Options options) {
         OptionGroup optionGroup = new OptionGroup();
 
-        Option opt = new Option("b", "brokerAddr", true, "query acl config version for which broker");
+        Option opt = new Option("b", "brokerAddr", true, "query acl config for which broker");
```
Comment (tsunghanjacktsai) [inline tools/src/main/java/org/apache/rocketmq/tools/command/acl/GetAccessConfigSubCommand.java:55]: Hi @meateggmilk ,

Please help us correct the description into "query acl config for specified cluster" to align with the -c option. Thx.

**Review by tsunghanjacktsai — CHANGES_REQUESTED** [review]

**Review by caigy — CHANGES_REQUESTED** [review]
It seems that modification of `GetAccessConfigSubCommand.java` has nothing to do with redundant logic, you'd better change it in another pr.
Also, please submit an issue first and link your pr to that issue.

File: tools/src/main/java/org/apache/rocketmq/tools/command/acl/GetAccessConfigSubCommand.java (hunk 1 above)
Comment (meateggmilk) [reply tools/src/main/java/org/apache/rocketmq/tools/command/acl/GetAccessConfigSubCommand.java:55]: ok, done it

**Review by tsunghanjacktsai — APPROVED** [review]
I'll approve the PR this time, but as @caigy said, anything changed that is not related to the corresponding issue should be avoided next time. Thx for your contribution btw.

**Comment by meateggmilk** [comment]
> 
Got it, thx

**Pull request closed without merging**